### ISSUE TITLE
azurerm_kubernetes_cluster can't use SystemAssigned identity with None as the private_dns_zone_id

### ISSUE THREAD

**Issue by justin-sobanski** [terraform-providers/terraform-provider-azurerm#10952]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.14.4
+ provider registry.terraform.io/hashicorp/azurerm v2.51.0

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "aks" {
  for_each                = var.clusters
  name                    = each.key
  location                = var.location
  kubernetes_version      = lookup(each.value, "kubernetes_version", null)
  resource_group_name     = var.resource_group_name
  dns_prefix              = each.key
  private_cluster_enabled = true
  private_dns_zone_id     = "None"

  default_node_pool {
    name                = local.default_pools[each.key].name
    node_count          = local.default_pools[each.key].node_count
    max_pods            = local.default_pools[each.key].max_pods
    vm_size             = local.default_pools[each.key].vm_size
    enable_auto_scaling = local.default_pools[each.key].enable_auto_scaling
    availability_zones  = local.default_pools[each.key].availability_zones
    max_count = (
      local.default_pools[each.key].enable_auto_scaling
      ? (
        local.default_pools[each.key].max_count != null
        ? local.default_pools[each.key].max_count
        : local.default_pools[each.key].node_count * 2
      )
      : null
    )
    min_count = (
      local.default_pools[each.key].enable_auto_scaling
      ? (
        local.default_pools[each.key].min_count != null
        ? local.default_pools[each.key].min_count
        : local.default_pools[each.key].node_count
      )
      : null
    )
    vnet_subnet_id = var.subnet_ids[each.value.subnet_name]
  }

  network_profile {
    network_plugin    = "azure"
    network_policy    = lookup(each.value, "network_policy", "calico")
    load_balancer_sku = "Standard"
    outbound_type     = "userDefinedRouting"
  }

  identity {
    type = "SystemAssigned"
  }

  addon_profile {
    kube_dashboard {
      enabled = false
    }
    oms_agent {
      enabled                    = var.log_analytics_workspace_id != null ? true : false
      log_analytics_workspace_id = var.log_analytics_workspace_id
    }
  }

  role_based_access_control {
    enabled = true
    azure_active_directory {
      managed                = true
      admin_group_object_ids = [azuread_group.adm[each.key].id]
    }
  }
  tags = local.tags
}
```

### Expected Behaviour

The cluster creates using `SystemAssigned` as the identity type. Since the DNS is handled outside of Azure and the cluster shouldn't need to manage it, `SystemAssigned` should be acceptable.

### Actual Behaviour

An error is received indicating that we need to use a user assigned identity or service principal:

`Error: a user assigned identity or a service principal must be used when using a custom private dns zone`

### Steps to Reproduce

1. `terraform apply` with `None` as the `private_dns_zone_id` and `SystemAssigned` as the identity type.

**Comment by favoretti** [comment]
Hey, thank you for reporting this. From my tests - this is an API limitation. AKS needs to have the privileges to that zone and you run into chicken and egg. AKS won't be able to start up before it can create the records and you won't be able to fetch the identity to assign rights for it to the zone.

**Comment by justin-sobanski** [comment]
@favoretti Thanks for the quick reply. That does make sense when it's an Azure DNS zone, however in my particular use-case  DNS is hosted on-premise. Because of this, the cluster identity won't be used to create or manage any of the DNS records, I'll be creating those separately through a different provider.

Based on the AKS documentation I believe this is the use-case for the `None` option:

```
"None" means AKS will not create a Private DNS Zone. This requires you to Bring Your Own DNS Server and configure the DNS resolution for the Private FQDN. If you don't configure DNS resolution, DNS is only resolvable within the agent nodes and will cause cluster issues after deployment.
```

So I believe skipping this check would only be applicable for the `None` option but still applicable when specifying an Azure DNS zone.

**Comment by favoretti** [comment]
Ahh. Yeah, this makes total sense. Didn't think of this usecase sorry. Ill shoot a PR in next week to fix this.

**Comment by favoretti** [comment]
Fixed by #11344
